ibm/service/vpc/data_source_ibm_is_reservation.go

```diff
@@ -0,0 +1,430 @@
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The resource type.",
+						},
+					},
+				},
+			},
+			"resource_group": &schema.Schema{
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "The resource group for this reservation.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"href": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The URL for this resource group.",
+						},
+						"id": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The unique identifier for this resource group.",
+						},
+						"name": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The user-defined name for this resource group.",
+						},
+					},
+				},
+			},
+			"resource_type": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The type of resource referenced.",
+			},
+			"status": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the reservation.",
+			},
+			"status_reasons": &schema.Schema{
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "The reasons for the current status (if any).",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"code": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "A snake case string succinctly identifying the status reason.",
+						},
+						"message": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "An explanation of the status reason.",
+						},
+						"more_info": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Optional:    true,
+							Description: "Link to documentation about this status reason.",
+						},
+					},
+				},
+			},
+			"zone": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The zone for this reservation.",
+			},
+		},
+	}
+}
+
+func dataSourceIBMIsReservationRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	var reservation *vpcv1.Reservation
+
+	if v, ok := d.GetOk("identifier"); ok {
+		id := v.(string)
+		getReservationOptions := &vpcv1.GetReservationOptions{}
+		getReservationOptions.SetID(id)
+		reservationInfo, response, err := sess.GetReservationWithContext(context, getReservationOptions)
+		if err != nil {
+			log.Printf("[DEBUG] GetReservationWithContext failed %s\n%s", err, response)
+			return diag.FromErr(fmt.Errorf("[ERROR] GetReservationWithContext failed %s\n%s", err, response))
+		}
+		reservation = reservationInfo
+
+	} else if v, ok := d.GetOk("name"); ok {
+
```

Comment:
Do we have validator to validate id or name as mutually exclusive for filter

Reply:
We get below error on terraform apply.
identifier": only one of `identifier,name` can be specified, but `identifier,name` were specified.

Removed else, kept it as separate if statement.